ocs_ci/utility/azure_utils.py

```diff
@@ -1,51 +1,180 @@
+# -*- coding: utf8 -*-
+"""
+Module for interactions with OCP/OCS Cluster on Azure platform level.
+"""
+
+import json
 import logging
+import os
 
 from azure.common.credentials import ServicePrincipalCredentials
 from azure.mgmt.compute import ComputeManagementClient
-from ocs_ci.utility.utils import TimeoutSampler
+from azure.mgmt.resource import ResourceManagementClient
+
+
+from ocs_ci.framework import config
 from ocs_ci.ocs.exceptions import TimeoutExpiredError
+from ocs_ci.utility.utils import TimeoutSampler
 
 logger = logging.getLogger(name=__file__)
 
 
+# default location of files with necessary azure cluster details
+SERVICE_PRINCIPAL_FILEPATH = os.path.expanduser("~/.azure/osServicePrincipal.json")
+TERRRAFORM_FILENAME = "terraform.azure.auto.tfvars.json"
+
+
+def load_cluster_resource_group(
+    cluster_path,
+    terraform_filename=TERRRAFORM_FILENAME
+):
+    """
+    Read terraform tfvars.json file created by ``openshift-installer`` in a
+    cluster dir to get azure ``resource group`` of an OCP cluster. All Azure
+    resources of the cluster are placed in this group.
+
+    Args:
+        cluster_path (str): full file path of the openshift cluster directory
+        terraform_filename (str): name of azure terraform vars file, this is
+            optional and you need to specify this only if you want to override
+            the default
+
+    Returns:
+        string with resource group name
+    """
+    filepath = os.path.join(cluster_path, terraform_filename)
+    with open(filepath, "r") as tf_file:
+        tf_dict = json.load(tf_file)
+    resource_group = tf_dict.get('azure_network_resource_group_name')
+    logger.debug(
+        "fetching azure resource group (%s) from %s file",
+        tf_dict.get('clientId'),
+        filepath
+    )
+    return resource_group
+
+
+def load_service_principal_dict(filepath=SERVICE_PRINCIPAL_FILEPATH):
+    """
+    Load Azure Service Principal from osServicePrincipal.json file and parse it
+    into a dictionary.
+
+    Args:
+        filepath (str): path of the
+
+    Returns:
+        dictionary with the service principal details (3 IDs and 1 secret)
+    """
+    with open(filepath, "r") as sp_file:
+        sp_dict = json.load(sp_file)
+    logger.debug(
+        "fetching azure service principal (clientId %s) from %s file",
+        sp_dict.get('clientId'),
+        filepath
+    )
+    return sp_dict
+
+
+# TODO: rename to AzureUtil
```

Comment:
Can we rename this now or is there something blocking that?

Reply:
There is a similar issue with AWS util class, so right now I'm just following the current convention. I will propose renaming AWS -> AWSUtil and AZURE to AzureUtil in a separate pull request (some code using these classes are importing it under such names already), as I tried leave low priority changes out of this pull request.

Reply:
Sounds good to me!